src/Analyser/MutatingScope.php

```diff
@@ -3820,6 +3838,31 @@ public function filterBySpecifiedTypes(SpecifiedTypes $specifiedTypes): self
 		foreach ($typeSpecifications as $typeSpecification) {
 			$expr = $typeSpecification['expr'];
 			$type = $typeSpecification['type'];
+
+			if ($expr instanceof IssetExpr) {
+				$issetExpr = $expr;
+				$expr = $issetExpr->getExpr();
+				$certainty = $issetExpr->getCertainty();
```

Comment:
This shouldn't be done through a property on IssetExpr, but rather this should work through `$typeSpecification['sure']` being true or false same as for all the other things.

Reply:
that was a great hint. I would have never thought this should be done via `$typeSpecification['sure']`